source/develop/release-management/features/gluster/gdeploy-cockpit-integration.html.md

```diff
@@ -0,0 +1,140 @@
+
+### UI Flow
+
+ HCI Deployment flow can be triggered by the option 'HCI Deployment with Gluster' in HE section of Ovirt plugin. This will open the following wizard which
+ help to provide all the required information about the gluster deployment.
+
+
+  ![Deploy-With-Glusterfs](Deploy-With-Gluster.png)
+
+
+ This wizard will have following five steps.
+
+ * Hosts:
+   This will help to capture all host information like IP address in the gluster cluster.
+
+
+  ![gdeploy-Hosts](gdeploy-hosts.png)
+
+
+* Packages:
+  This is used for capturing the packages which needs to be installed. This can also take care of additional repos which needs to enabled to install those packages.
+  In case of RHEL systems, this can also takes care of registering the system with CDN and enabling the required repos.
+
+  ![gdeploy-Packages](gdeploy-packages.png)
+
+* Volumes:
+  This step is used to capture the all the gluster volume details which needs to be created by gdeploy
+
+
+  ![gdeploy-Volumes](gdeploy-volumes.png)
+
+
+* Bricks:
+  This step will be used to define the bricks for the glusterfs Volumes defined in previous step.
+
+
+  ![gdeploy-bricks](gdeploy-bricks.png)
+
+
+* Summary:
+  This will be the final step in the gdeploy flow and it will show a summary of all the inputs captured in the previous steps.
+  This will generate a gdeploy config file with all the inputs captured in previous step. This step will also create an answer file for Hosted-Engine deployment with default answers for HCI deployment and gluster volume information for hosted-engine storage domain.
+  User can review the answer files and click 'Finish' to start executing the gluster deployment using gdeploy. Progress of gdeploy execution will be shown as it executes different steps in the deployment.
+
+
+    ![gdeploy-summary](gdeploy-summary.png)
+
+
+    ![gdeploy-log](gdeploy-log.png)
+
+
+* Hosted-Engine Setup:
+  After completing the gluster deployment, it will start the standard Hosted-Engine deployment by invoking the 'hosted-engine --deploy' with the generated answer file. Standard HE deployment work flow will start from this step.
+
+
+   ![he-deploy](he-deploy.png)
+
+
+## Implementation
+
+We need following changes to implement this feature.
+
+### UI Changes:
+
+* Provide a option in 'Hosted-Engine' section to start the Hosted-Engine deployment with Gluster. This should invoke the gdeploy
+wizard.
+
+* Gdeploy wizard shown in previous section needs to implemented using react-js. At the final step, this wizard should call a script(python?) using cockpit.spawn() to generate the config files for 'gdeploy' and
+'hosted-engine'. Once user reviews and confirms the generated config files then execute the 'gdeploy' command using cockpit.spawn('gdeploy -c gluster.conf'). UI should track the execution and report the results as and when different steps are being executed by gdeploy.
```

Comment:
I am not aware of cockpit.file. But as of now, we don't have the utility to create gdeploy conf file. As you suggested, let me explore  cockpit.file and make use of it directly instead of python.